core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -153,6 +178,15 @@ export class ResultPanelComponent {
   }
 
   public clearResultPanel(): void {
+    // store result into session storage so that they could be restored
+    //   when user click the "view result" operator again
+    if (sessionStorage.getItem('newWorkflowExecuted') === 'false' && this.currentResult.length > 0) {
+      sessionStorage.setItem('currentResult', JSON.stringify(this.currentResult));
+      sessionStorage.setItem('currentPageIndex', JSON.stringify(this.currentPageIndex));
+      sessionStorage.setItem('currentPageSize', JSON.stringify(this.currentPageSize));
+      sessionStorage.setItem('total', JSON.stringify(this.total));
```

Comment:
move session storage operations to separate functions, see comment above

Reply:
I suggest we can use the same util code from #915

Reply:
In order to use the utility @Yicong-Huang mentioned, I merged the master into my branch.

However, I found that currently there is no `version.ts` in the `new-gui/src/environments` folder, causing error in `navigation.component.ts` and preventing the frontend from running.

I examined the history of `navigation.component.ts` on master branch and found that in [this commit `f8f5f79`](https://github.com/Texera/texera/commit/f8f5f79942256641a812ce28052dbdf6b36be8a9#diff-a2405c8da3a8489eb535b4147bb55dbf78746dac75b9e188ad2c2d0953eab5a0), the `version.ts` was added. However, according to the file content, the `version.ts` was auto generated. How do I generate this file?

Reply:
@WANGJIEKE please do an npm install

Reply:
Updated in commit https://github.com/Texera/texera/pull/903/commits/d668ef2d6eb6c5ef694c00eeaa984ccf51bd860e